packages/api/src/external/carequality/patient-updater-carequality.ts

```diff
@@ -42,9 +40,9 @@ export class PatientUpdaterCarequality extends PatientUpdater {
     const updatePatient = async (patient: Patient) => {
       try {
         const facilityId = getFacilityIdOrFail(patient);
-        const facility = await getFacilityOrFail({ cxId, id: facilityId });
-        const requestId = uuidv7();
-        await cqCommands.patient.discover(patient, facility.id, requestId);
+        await getFacilityOrFail({ cxId, id: facilityId });
+        // WARNING Could interfere with currently running state of another PD
```

Comment:
has this warning been resolved? like is this the current state of the PR?

Reply:
It hasn't been resolved, but it's also not a huge problem at the moment per se. Right now, if you call `updateAll`, the `discover` calls for the patients will fire off even if there's an on-going patient discovery run for a given patient. Given we have a single PD state per patient per HIE in both `patient` and `cq_patient_data`, there's a risk both runs start editing the same state.

The reason it's maybe not the end of the world (i.e WARNING right now), is that I don't think multiple `discover` runs rely on the patient state for critical things. Each new run will have its own request ID and I think stays siloed when calculating links. There *may* be unintended outcomes in `cq_patient_data` depending on who updates that first.

More explicitly, for analytics, we rely on saving the start date in the patient state, which could get overwritten and produce an inaccurate duration. A side-effect today but may cause trouble down the road.

Reply:
It's the same issue I noted in the `I made this a BUG not WARNING...` comment you responded to.

Reply:
Will likely capture this in a larger PD state issue.

Reply:
Per discussion: will leave warnings in place for now.